tests/python/test_predict.py

```diff
@@ -231,6 +231,7 @@ def test_base_margin(self):
         from_dmatrix = booster.predict(dtrain)
         np.testing.assert_allclose(from_dmatrix, from_inplace)
 
+    @pytest.mark.skipif(**tm.no_pandas())
```

Comment:
Is this necessary?

Reply:
Yes. The test also uses pandas in `np_dtypes` for testing differences between np.dtype and pandas types. For instance, what happen if np.bool is passed in? Is it nullable in a pandas dataframe?

Reply:
Still not sure. Neither `np_dtypes` nor `test_dtypes` has any reference to Pandas?

Reply:
https://github.com/dmlc/xgboost/blob/0c38ca7f6e329dbb062939dcb843d0dc229675b4/python-package/xgboost/testing/data.py#L46